controllers/machinehealthcheck_controller.go

```diff
@@ -494,7 +495,7 @@ func (r *MachineHealthCheckReconciler) machineToMachineHealthCheck(o client.Obje
 	var requests []reconcile.Request
 	for k := range mhcList.Items {
 		mhc := &mhcList.Items[k]
-		if hasMatchingLabels(mhc.Spec.Selector, m.Labels) {
+		if machine.HasMatchingLabels(mhc.Spec.Selector, m.Labels) {
```

Comment:
Just a question / future concern for now, this change is fine for me for this PR: (just to surface it already)

When the other controllers are moved too: do we allow dependencies between the different controller implementations or do we expect all shared code to reside in separate shared (util) packages? (we can only introduce dependencies between controller packages until we hit a cyclic dependency)

This wasn't a concern with the old structure as everything was in the same package anyway.

In my opinion we should avoid dependencies between different controller packages mid-term/long-term to avoid complexity in the package structure/dependencies.

Reply:
Just saw that we have dependencies like this for the tests anyway, but we should maybe try to only do this in tests.

Anyway, we have to be careful that we don't run into cyclic dependencies, e.g. this means (afaik) that e.g. the cluster controller can also use the machine controller in tests, but then the machine controller cannot use the cluster controller too.

As long as we have a clear dependency  "hierarchy" between controllers this should be fine, otherwise we will run into a problem in later stages of this refactoring. Maybe we should check this already (not sure if you already have a local final version which still compiles)

Reply:
I'm not happy about those cross dependencies either but I'm trying to keep this move as simple as possible
But I have a local version with the full refactor and we should not have problems

Reply:
> 

Okay perfect, then let's just resolve for now. Enough for me that it's not a showstopper for the upcoming moves and that we're aware of that concern.